modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-fieldset/src/main/resources/META-INF/resources/fieldset_field.js

```diff
@@ -22,19 +34,168 @@ AUI.add(
 
 				prototype: {
 
+					initializer: function() {
+						var instance = this;
+
+						if (instance.get('repeatedIndex') > 0) {
+							instance.set('showLabel', false);
+						}
+
+						var context = instance.get('context');
+
+						context.fields.forEach(function(fieldContext) {
+							fieldContext.fieldName = Util.getFieldNameFromQualifiedName(fieldContext.name);
+
+							var field = instance.getField(fieldContext.fieldName);
+
+							if (!fieldContext.instanceId) {
+								var instanceId = Util.generateInstanceId(8);
+
+								fieldContext.instanceId = instanceId;
```

Comment:
Do we need to set `instanceId` here? I though it would automatically get generated when doing `new Field()`